### PR TITLE
Fix sorting of Unicode output paths in RE API protos

### LINKED ISSUE #26745: remote client sorts strings by UTF-16 representation rather than code point
The REAPI generally mandates repeated string fields in its protobuf messages be sorted by code point or equivalently UTF-8 representation. Bazel tends to use the natural ordering of `String` in Java to sort, which is not equivalent to ordering by code point.

For example, this test demonstrates that `Command.output_paths` are sorted improperly:
```
  @Test
  public void buildRemoteAction_SMP() throws Exception {
    var one = "path/ﾐ";
    var oneF = execRoot.getRelative(unicodeToInternal(one)).asFragment();
    var two = "path/🔥";
    var twoF = execRoot.getRelative(unicodeToInternal(two)).asFragment();
    var spawn =
        new SpawnBuilder("dummy")
            .withOutput(ActionsTestUtil.createArtifactWithExecPath(artifactRoot, oneF))
            .withOutput(ActionsTestUtil.createArtifactWithExecPath(artifactRoot, twoF))
            .build();
    FakeSpawnExecutionContext context = newSpawnExecutionContext(spawn);
    RemoteExecutionService service = newRemoteExecutionService();

    RemoteAction remoteAction = service.buildRemoteAction(spawn, context);

    assertThat(remoteAction.getCommand().getOutputPathsList())
        .containsExactly("/execroot/" + one, "/execroot/" + two)
        .inOrder();
  }
```
Indeed, ﾐ (U+FF90) < 🔥 (U+1F525).

### PR DESCRIPTION
**Pull request by fmeum**
These paths were sorted as Java UTF-16 strings, not as UTF-8 byte sequences as required by the RE API spec.

Also add test coverage for Unicode characters to the golden test and verify that output paths are sorted as strings, not paths.

Fixes #26745

### CODE REVIEW

**Comment by fmeum** [comment]
@bazel-io fork 8.4.0

File: src/main/java/com/google/devtools/build/lib/remote/RemoteExecutionService.java [hunk 1]
```diff
@@ -171,6 +173,9 @@
  * cache and execution with spawn specific types.
  */
 public class RemoteExecutionService {
+  private static final Comparator<String> PROTO_STRING_COMPARATOR =
+      comparing(StringEncoding::unicodeToInternal, String::compareTo);
```
Comment (tjgq) [inline src/main/java/com/google/devtools/build/lib/remote/RemoteExecutionService.java]: Isn't the 2nd argument redundant?

Comment (fmeum) [inline src/main/java/com/google/devtools/build/lib/remote/RemoteExecutionService.java]:
```suggestion
      comparing(StringEncoding::unicodeToInternal);
```

**Review by tjgq — APPROVED** [review]

**Comment by iancha1992** [comment]
@fmeum @tjgq Looks like there is a `bazel_windows_cpp_test` error

**Comment by fmeum** [comment]
@iancha1992 That's a failure at HEAD that has since been fixed, I updated the branch.
